pandas/tests/arrays/test_datetimes.py

```diff
@@ -16,6 +16,25 @@
 
 
 class TestDatetimeArrayConstructor(object):
+    def test_mixing_naive_tzaware_raises(self):
+        # GH#24569
+        arr = np.array([pd.Timestamp('2000'), pd.Timestamp('2000', tz='CET')])
+
+        msg = ('Cannot mix tz-aware with tz-naive values|'
+               'Tz-aware datetime.datetime cannot be converted '
+               'to datetime64 unless utc=True')
+
+        for obj in [arr, arr[::-1]]:
+            # check that we raise regardless of whether naive is found
+            #  before aware or vice-versa
+            for meth in [DatetimeArray._from_sequence,
+                         sequence_to_dt64ns,
+                         pd.to_datetime,
+                         pd.DatetimeIndex]:
```

Comment:
Can you parametrize over `meth` instead of iterating?

Reply:
I can if you have a strong preference.  I'm trying to push back against over-parametrizing, since pytest overhead is becoming noticeable.

Reply:
i think it’s worth it here for ease of debuggjng

Reply:
done